microsoft/CBL-Mariner
# Add flag to disable PMC repository

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*

---

###### Summary 

- The `REPO_LIST` variable defined in our makefile is used to specify package repositories to be used by `tdnf` in addition to existing PMC repository. Add an option to explicitly disable PMC so that `tdnf` will only use the repos specified in `REPO_LIST`.

###### Change Log  



- Add new flag in Makefile `DISABLE_MARINER_REPOS` which defaults to false
- Add logic to `rpmrepocloner` to disable PMC repos when the flag is passed

###### Does this affect the toolchain?  



**NO**

###### Test Methodology

- Local build
  * Create a local file `custom.repo`, which is a copy of https://github.com/microsoft/CBL-Mariner/blob/2.0/SPECS/mariner-repos/mariner-official-base.repo
  * Run a make command to build packages or create image with `DISABLE_MARINER_REPOS=y`. The command will fail because `tdnf` cannot find some packages.
  * Run the same command with an additional variable `REPO_LIST="/path/to/custom.repo"`. The build command should work.
